sigmastate/src/main/scala/sigmastate/utxo/transformers.scala

```diff
@@ -271,7 +271,7 @@ case class ByIndex[V <: SType](input: Value[SCollection[V]],
     default match {
       case Some(d) =>
         val dV = d.evalTo[V#WrappedType](env)
-        Value.checkType(d, dV) // necessary because cast to V#WrappedType is erased
+        Value.checkType(d.tpe, dV) // necessary because cast to V#WrappedType is erased
```

Comment:
It is secure, because checkType has overload for SType. 
Rolled back, because passing just d will have more detailed error message.
In case you wonder why overload is necessary, it is because of this place https://github.com/ScorexFoundation/sigmastate-interpreter/blob/81300fd7df1d98d37df5629f672b52dc413c8004/sigmastate/src/main/scala/sigmastate/Values.scala#L1178
where there is no Value.